package.json

```diff
@@ -5,8 +5,10 @@
     "setup": "npm i && lerna bootstrap",
     "start": "lerna run --parallel start",
     "build": "lerna run build",
-    "test": "lerna run test",
+    "test": "lerna run test && npm run prettier:check && npm run lint:check",
```

Comment:
Why not run these checks as part of `lint:check` below? This is the command used in the jenkinsfile.

Also, I'm a little lost as to how the linting is working now. I see eslint, and now prettier. Do we need both or can we just use one and remove the other?

Reply:
They are, at least IMHO two separate things:
 - ESLint stands for EcmaScript Lint and is focused on ensuring good code practice in JavaScript and its derivate. Code practice means that there's an impact on the _logic_ of the code.
 - Prettier is here to ensures that everything that it supports abides by specific formatting. This means that there's no impact on the _logic_ of the code, it's only standardization of the format (because we're not using odd languages that rely on indentation and whatnot for their logic /s)
 
 So, both are needed because they both do different things

Reply:
You can see an example in #573. I didn't do anything related to atomic, but when I merged, the hook detects the file has changed and formatted them.

Prettier and ESLint works well only if everyone plays ball and that the rules are enforced.

Reply:
Ok, what about the first point on running all linting as part of the `lint:check` command? Why is the check being grouped with the `test` command?

Reply:
Let's say I work on `fileA` and `fileB` and you work on `fileB`. I merge my changes before yours, but I've been a naughty dev and did a `git commit --no-verify` and thus bypassed eslint and prettier. 

Now comes the time for you to do your PR. Oh, chucks, conflict on `fileB`. You merge main in your branch, fix the conflict and commit-merge (without no-verify).
Now you're PR is littered with code style changes, or worse, your merge commit failed because I ignored ESLint rules, which:
1. Hinder the review, making your reviewers lose time
2. Make you lose time because you have to either poke the bad dev or fix this yourself (which, incidentally, is what I'm doing here...)

Reply:
I appreciate the anecdote, and I agree we want to prevent these scenarios. However, the `lint:check` is [being run on CI](https://github.com/coveo/ui-kit/blob/master/Jenkinsfile#L25) in it's own stage. Couldn't we achieve the same solution by piggy-backing on the lint command?

Why should the `test` stage fail when the error is a linting one?

Reply:
Hm yeah I tend to see the linting as a test, personally. But I agree, I'll change that.